src/Native/System.Native/pal_exec.cpp

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "../config.h"
+#include "pal_exec.h"
+
+#include <assert.h>
+#include <errno.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+enum
+{
+    READ_END_OF_PIPE = 0,
+    WRITE_END_OF_PIPE = 1,
+};
+    
+inline static void CloseIfOpen(int fd)
+{
+    if (fd >= 0)
+        close(fd);
+}
+
+extern "C"
```

Comment:
:) I've gone in the other direction and removed the pasted comments from the header for the parameters (did not mean to duplicate them) so the impl signature looks more bare and hints more at needing to look at the header for docs...